### PR TITLE
rss: Better new entries check with hashing

### PR DESCRIPTION
**Pull request by mrshu**
* Changed the way new entries are being checked/found by comparing their
  hashes. The hash is composed of the following:

  - Entry ID
  - Entry Title
  - Entry URL

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

**Comment by mrshu** [comment]
to @pepol @Adman

File: rss.py [hunk 1]
```diff
@@ -17,27 +18,32 @@ def setup(self, *args, **kwargs):
 
         for feed in self.feeds:
             if feed not in self.storage:
-                self.storage[feed] = ''
+                self.storage[feed] = []
         self.loop_task(self.update_interval, self.check_feeds, now=False)
 
+    def hash(self, entry):
+        return hashlib.sha224("{}{}{}".format(entry.id,
```
Comment (pepol) [inline rss.py]: Wasn't the original problem with entries that had different IDs while their content was the same? As far as I can see, including the entry ID would thus not solve the original problem, or am I mistaken?

Also, why sha224? :-D

Comment (mrshu) [inline rss.py]: @pepol sha224 just looks cool =)

You are right on the ID, not sure how I missed that but in my tests this seemed to behave in a pretty stable manner.

In any case, I personally think we are only interested in the title and the link anyway so I am ditching the ID.

**Comment by mrshu** [comment]
@pepol would you mind taking another look?

**Comment by pepol** [comment]
LGTM, feel free to merge it.

**Pull request merged**